substrate/frame/collective/src/migrations/v5.rs

```diff
@@ -0,0 +1,200 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::{Config, Pallet};
+use codec::{Decode, Encode};
+use frame_support::{
+	pallet_prelude::{OptionQuery, StorageVersion, TypeInfo, ValueQuery},
+	sp_runtime::RuntimeDebug,
+	traits::{Get, Hash as PreimageHash, OnRuntimeUpgrade, QueryPreimage, StorePreimage},
+	weights::Weight,
+	BoundedVec, Identity,
+};
+use frame_system::pallet_prelude::BlockNumberFor;
+use sp_std::prelude::*;
+
+#[frame_support::storage_alias]
+pub type Proposals<T: Config<I>, I: 'static> = StorageValue<
+	Pallet<T, I>,
+	BoundedVec<<T as frame_system::Config>::Hash, <T as Config<I>>::MaxProposals>,
+	ValueQuery,
+>;
+
+#[frame_support::storage_alias]
+pub type ProposalOf<T: Config<I>, I: 'static> = StorageMap<
+	Pallet<T, I>,
+	Identity,
+	<T as frame_system::Config>::Hash,
+	<T as Config<I>>::Proposal,
+	OptionQuery,
+>;
+
+#[frame_support::storage_alias]
+pub type Voting<T: Config<I>, I: 'static> = StorageMap<
+	Pallet<T, I>,
+	Identity,
+	<T as frame_system::Config>::Hash,
+	OldVotes<<T as frame_system::Config>::AccountId, BlockNumberFor<T>>,
+	OptionQuery,
+>;
+
+#[frame_support::storage_alias]
+pub type Members<T: Config<I>, I: 'static> =
+	StorageValue<Pallet<T, I>, Vec<<T as frame_system::Config>::AccountId>, ValueQuery>;
+
+/// Info for keeping track of a motion being voted on.
+#[derive(PartialEq, Eq, Clone, Encode, Decode, RuntimeDebug, TypeInfo)]
+pub struct OldVotes<AccountId, BlockNumber> {
+	/// The proposal's unique index.
+	pub index: crate::ProposalIndex,
+	/// The number of approval votes that are needed to pass the motion.
+	pub threshold: crate::MemberCount,
+	/// The current set of voters that approved it.
+	pub ayes: Vec<AccountId>,
+	/// The current set of voters that rejected it.
+	pub nays: Vec<AccountId>,
+	/// The hard end time of this vote.
+	pub end: BlockNumber,
+}
+
+pub struct MigrateToV5<T, I>(sp_std::marker::PhantomData<(T, I)>);
+impl<T: Config<I>, I: 'static> OnRuntimeUpgrade for MigrateToV5<T, I> {
+	#[cfg(feature = "try-runtime")]
+	fn pre_upgrade() -> Result<Vec<u8>, sp_runtime::TryRuntimeError> {
+		log::info!("pre-migration collective v5");
+		frame_support::ensure!(
+			StorageVersion::get::<Pallet<T, I>>() < 5,
+			"collective already migrated"
+		);
```

Comment:
Can you please use the `VersionedMigration` struct? So we don't need any version handling logic in the migration